src/OrderManager/V7/OrderManager.php

```diff
@@ -307,10 +310,21 @@ public function recreateOrder(CartInterface $cart): AbstractOrder
             $order->setOrderdate(new Carbon());
             $order->setCartId($sourceOrder->getCartId());
 
+            if ($tokens) {
+                $tokenVersionNote = ' Token previously added but removed: ';
+                foreach ($tokens as $token) {
+                    $this->voucherService->removeAppliedTokenFromOrder($token, $sourceOrder);
+                    $this->voucherService->applyToken($token->getToken(), $cart, $order);
```

Comment:
Is this required as the tokens are already applied in `getOrCreateOrderFromCart`

Reply:
mhh, right, it should be `$cart->addVoucherToken($token->getToken());` otherwise is doing the application twice

Reply:
https://github.com/pimcore/ecommerce-framework-bundle/pull/121/commits/e56cbf7dba23269883de65fc5fa9947afb8192a8 Theoretically it's not required to apply anything again on the new order, just "unusing" it would be fine, it is still in the same Cart which stores the token, and once getting into `getOrCreateOrderFromCart` it would see the same voucher but as unused and re-apply.

Limiting this functionality to all orders that don't have the status `commited` which is the final stage(?) and it means that the token is considered to be already consumed.